### PR TITLE
Specify in guidelines one should split CPUs when module has n > 1 tool

### PR DESCRIPTION
**Pull request by jfy133**
Slack reference: https://nfcore.slack.com/archives/CJRH30T6V/p1628512208312100

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -457,6 +457,8 @@ using a combination of `bwa` and `samtools` to output a BAM file instead of a SA
 
 - If the tool supports multi-threading then you MUST provide the appropriate parameter using the Nextflow `task` variable e.g. `--threads $task.cpus`.
 
+- If a module contains _two_ tools that supports multi-threading (e.g. [piping output into a samtools command](https://github.com/nf-core/modules/blob/28b023e6f4d0d2745406d9dc6e38006882804e67/modules/bowtie2/align/main.nf#L32-L46)), split `task.cpus` into a new variable that are given to both tools, allowing both to utilise the provided CPUs.
```
Comment (grst) [inline README.md:460]:
```suggestion
- If a module contains _multiple_ tools that supports multi-threading (e.g. [piping output into a samtools command](https://github.com/nf-core/modules/blob/28b023e6f4d0d2745406d9dc6e38006882804e67/modules/bowtie2/align/main.nf#L32-L46)), you MUST assign cpus per tool such that the total number of used CPUs does not exceed `task.cpus`. 
```

I didn't get the part about the variable... Also it doesn't necessarily make sense to split cpus 50:50?

Comment (jfy133) [reply README.md:460]: Good point, (once again) PR was in response to one issue on one module PR.

By the variable I'm referring specifically to the [`split_cpus` variable](https://github.com/nf-core/modules/blob/28b023e6f4d0d2745406d9dc6e38006882804e67/modules/bowtie2/align/main.nf#L32)  in the example. I 'm assuming it makes sense to have that consistent across all modules.

**Comment by grst** [comment]
I was also thinking of [this case](https://github.com/nf-core/modules/blob/6e68c1af9a514bb056c0513ebba6764efd6750fc/modules/bwa/sampe/main.nf#L42-L43). But you are right, there are a bunch of modules now that use `split_cpus = Math.floor(task.cpus/2)` now.

**Comment by jfy133** [comment]
Ok then I add an extra example if with use two, and alternatively the `-1` doodah

**Comment by Emiller88** [comment]
Have we done a side by side test to confirm that splitting the CPUs is the most efficient way?

**Comment by grst** [comment]
what would be the alternative? Overbooking cpu resources is bad practice as not all schedulers restrict cpu usage to the requested number.

**Review by grst — APPROVED** [review]
LGTM now

**Pull request merged**